messaging/broker.go

```diff
@@ -774,6 +791,33 @@ func (t *topic) Close() error {
 	return nil
 }
 
+// loadIndex reads the highest available index for a topic from disk.
+func (t *topic) loadIndex() error {
```

Comment:
When we add truncation (and log segments) then it'll only have to traverse through the latest log segment for each topic.